controller/volume_controller.go

```diff
@@ -560,11 +560,13 @@ func (vc *VolumeController) ReconcileVolumeState(v *longhorn.Volume, e *longhorn
 	// it needs to be attached automatically so that its engine will be launched then
 	// restore data from backup.
 	if v.Spec.InitialRestorationRequired {
-		usableNode, err := vc.ds.GetRandomReadyNode()
-		if err != nil {
-			return err
+		if v.Spec.NodeID == "" {
+			usableNode, err := vc.ds.GetRandomReadyNode()
+			if err != nil {
+				return err
+			}
+			v.Spec.NodeID = usableNode.Name
```

Comment:
`DisableFrontend` should be a part of the clause.

So `if v.Spec.InitialRestorationRequired && v.Spec.NodeID == ""`